lib/boa/type.rb

```diff
@@ -176,5 +180,37 @@ def freeze
 
       T.let(super(), Type)
     end
+
+    # Parse the value
+    #
+    # @example when the value is included
+    #   type = String.new(:first_name, includes: %w[Jon])
+    #   type.parse('Jon') # => Boa::Success.new('Jon')
+    #
+    # @example when the value is not included
+    #   type = String.new(:first_name, includes: %w[Jon])
+    #   type.parse('Dan') # => Boa::Failure.new('must be one of ["Jon"], but was "Dan"')
+    #
+    # @example when includes is nil
+    #   type = String.new(:first_name)
+    #   type.parse('Jon') # => Boa::Success.new('Jon')
+    #
+    # @param value [Object] the value to parse
+    #
+    # @return [Result<Object, String>] the result of parsing the value
+    #
+    # @api public
+    sig { overridable.params(value: ::Object).returns(Result[T.untyped, ExceptionType]) }
+    def parse(value)
+      # TODO: Check to see if value matches primitive
+
+      allowed_values = includes
```

Comment:
__comment:__ This is required so that sorbet can trace the *variable* and see it's tested with `#nil?` first, then `#include?` is permitted. Without this, sorbet will not assume that `#includes` returns the same value each time it is executed.